src/interned.rs

```diff
@@ -137,7 +137,25 @@ where
 
         // Optimisation to only get read lock on the map if the data has already
         // been interned.
+        let mut hasher = self.key_map.hasher().build_hasher();
+        data.hash(&mut hasher);
+        let data_hash = hasher.finish();
+        let shard = self.key_map.determine_shard(data_hash as _);
+        {
+            let lock = self.key_map.shards()[shard].read();
+            if let Some(bucket) = lock.find(data_hash, |(a, _)| {
+                // lifetime shrink
+                let a: &C::Data<'db> = unsafe { std::mem::transmute(a) };
```

Comment:
```suggestion
                let a: &C::Data<'db> = unsafe { self.to_internal_data(a) };
```

Reply:
I need from_internal_data here because i don't want to use GAT for Owned type in Lookup trait, what is necessary to express eq(&Key<'db>, &Data<'any_lifetime>), but it requires &'db C::Data and we have only &'1 C::Data here

Reply:
(Too annoying?)

Reply:
Also, yeah, I had it wrong, I see that it should be `from_internal_data`....do we not have this method though?

Reply:
Yes, GAT bring us to more annoying bounds. Self::from_internal_data requires &'db C::Data<'db> and idk how to express this bound on eq callback